extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchUnaryPostAggregator.java

```diff
@@ -70,7 +70,7 @@ public boolean equals(final Object o)
     if (!super.equals(o)) {
       return false;
     }
-    if (!(o instanceof ArrayOfDoublesSketchUnaryPostAggregator)) {
+    if (o == null || getClass() != o.getClass()) {
```

Comment:
Why this change?

Reply:
I believe you asked for it:

> I was wrong when I've written "Then it breaks the substitution principle", because the superclass is abstract. Actually we should preserve getClass() in abstract classes in this hierarchy, so that instances of different concrete subclasses that don't override equals() don't appear "equal" to each other.

So I added tests for equals() methods and made this change.

Reply:
added